// @headlessui-float/react/dist/headlessui-float.js
import ce, { useLayoutEffect as xt, useEffect as te, useState as le, createContext as St, forwardRef as Ot, useRef as jt, useCallback as Pt, isValidElement as Je, Fragment as q, useMemo as Ft, useContext as kt } from "react";
import { createPortal as At } from "react-dom";
import { Transition as It } from "@headlessui/react";
import { useFloating as $t, offset as Dt, flip as Wt, shift as Mt, autoPlacement as Nt, arrow as Ut, hide as Lt, autoUpdate as Yt } from "@floating-ui/react-dom";
var ue = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, Ke = "Expected a function", Ve = 0 / 0, Vt = "[object Symbol]", Bt = /^\s+|\s+$/g, Ht = /^[-+]0x[0-9a-f]+$/i, zt = /^0b[01]+$/i, qt = /^0o[0-7]+$/i, Gt = parseInt, Jt = typeof ue == "object" && ue && ue.Object === Object && ue, Kt = typeof self == "object" && self && self.Object === Object && self, Xt = Jt || Kt || Function("return this")(), Zt = Object.prototype, Qt = Zt.toString, er = Math.max, tr = Math.min, we = function() {
  return Xt.Date.now();
};
function rr(t, f, i) {
  var h, R, x, E, b, l, w = 0, g = !1, _ = !1, S = !0;
  if (typeof t != "function")
    throw new TypeError(Ke);
  f = Be(f) || 0, de(i) && (g = !!i.leading, _ = "maxWait" in i, x = _ ? er(Be(i.maxWait) || 0, f) : x, S = "trailing" in i ? !!i.trailing : S);
  function T(c) {
    var v = h, k = R;
    return h = R = void 0, w = c, E = t.apply(k, v), E;
  }
  function M(c) {
    return w = c, b = setTimeout(F, f), g ? T(c) : E;
  }
  function U(c) {
    var v = c - l, k = c - w, N = f - v;
    return _ ? tr(N, x - k) : N;
  }
  function I(c) {
    var v = c - l, k = c - w;
    return l === void 0 || v >= f || v < 0 || _ && k >= x;
  }
  function F() {
    var c = we();
    if (I(c))
      return O(c);
    b = setTimeout(F, U(c));
  }
  function O(c) {
    return b = void 0, S && h ? T(c) : (h = R = void 0, E);
  }
  function L() {
    b !== void 0 && clearTimeout(b), w = 0, h = l = R = b = void 0;
  }
  function G() {
    return b === void 0 ? E : O(we());
  }
  function Y() {
    var c = we(), v = I(c);
    if (h = arguments, R = this, l = c, v) {
      if (b === void 0)
        return M(l);
      if (_)
        return b = setTimeout(F, f), T(l);
    }
    return b === void 0 && (b = setTimeout(F, f)), E;
  }
  return Y.cancel = L, Y.flush = G, Y;
}
function nr(t, f, i) {
  var h = !0, R = !0;
  if (typeof t != "function")
    throw new TypeError(Ke);
  return de(i) && (h = "leading" in i ? !!i.leading : h, R = "trailing" in i ? !!i.trailing : R), rr(t, f, {
    leading: h,
    maxWait: f,
    trailing: R
  });
}
function de(t) {
  var f = typeof t;
  return !!t && (f == "object" || f == "function");
}
function ar(t) {
  return !!t && typeof t == "object";
}
function ir(t) {
  return typeof t == "symbol" || ar(t) && Qt.call(t) == Vt;
}
function Be(t) {
  if (typeof t == "number")
    return t;
  if (ir(t))
    return Ve;
  if (de(t)) {
    var f = typeof t.valueOf == "function" ? t.valueOf() : t;
    t = de(f) ? f + "" : f;
  }
  if (typeof t != "string")
    return t === 0 ? t : +t;
  t = t.replace(Bt, "");
  var i = zt.test(t);
  return i || qt.test(t) ? Gt(t.slice(2), i ? 2 : 8) : Ht.test(t) ? Ve : +t;
}
var or = nr;
const Xe = typeof window < "u" ? xt : te, _e = { serverHandoffComplete: !1 };
function fr() {
  const [t, f] = le(_e.serverHandoffComplete);
  return te(() => {
    t !== !0 && f(!0);
  }, [t]), te(() => {
    _e.serverHandoffComplete === !1 && (_e.serverHandoffComplete = !0);
  }, []), t;
}
let ur = 0;
function He() {
  return ++ur;
}
var Ge;
const sr = (Ge = ce.useId) != null ? Ge : function() {
  const f = fr(), [i, h] = ce.useState(f ? He : null);
  return Xe(() => {
    i === null && h(He());
  }, [i]), i != null ? `${i}` : void 0;
}, yr = [
  "origin-bottom",
  "origin-top",
  "origin-right",
  "origin-left",
  "origin-bottom-left",
  "origin-bottom-right",
  "origin-top-left",
  "origin-top-right"
], lr = (t) => {
  switch (t) {
    case "top":
      return "origin-bottom";
    case "bottom":
      return "origin-top";
    case "left":
      return "origin-right";
    case "right":
      return "origin-left";
    case "top-start":
    case "right-end":
      return "origin-bottom-left";
    case "top-end":
    case "left-end":
      return "origin-bottom-right";
    case "right-start":
    case "bottom-start":
      return "origin-top-left";
    case "left-start":
    case "bottom-end":
      return "origin-top-right";
    default:
      return "";
  }
};
var Ze = { exports: {} }, Q = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ze;
function cr() {
  if (ze)
    return Q;
  ze = 1;
  var t = ce, f = Symbol.for("react.element"), i = Symbol.for("react.fragment"), h = Object.prototype.hasOwnProperty, R = t.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, x = { key: !0, ref: !0, __self: !0, __source: !0 };
  function E(b, l, w) {
    var g, _ = {}, S = null, T = null;
    w !== void 0 && (S = "" + w), l.key !== void 0 && (S = "" + l.key), l.ref !== void 0 && (T = l.ref);
    for (g in l)
      h.call(l, g) && !x.hasOwnProperty(g) && (_[g] = l[g]);
    if (b && b.defaultProps)
      for (g in l = b.defaultProps, l)
        _[g] === void 0 && (_[g] = l[g]);
    return { $$typeof: f, type: b, key: S, ref: T, props: _, _owner: R.current };
  }
  return Q.Fragment = i, Q.jsx = E, Q.jsxs = E, Q;
}
var ee = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var qe;
function dr() {
  return qe || (qe = 1, process.env.NODE_ENV !== "production" && function() {
    var t = ce, f = !1, i = !1, h = !1, R = !1, x = !1, E = Symbol.for("react.element"), b = Symbol.for("react.portal"), l = Symbol.for("react.fragment"), w = Symbol.for("react.strict_mode"), g = Symbol.for("react.profiler"), _ = Symbol.for("react.provider"), S = Symbol.for("react.context"), T = Symbol.for("react.forward_ref"), M = Symbol.for("react.suspense"), U = Symbol.for("react.suspense_list"), I = Symbol.for("react.memo"), F = Symbol.for("react.lazy"), O = Symbol.for("react.offscreen"), L = Symbol.iterator, G = "@@iterator";
    function Y(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = L && e[L] || e[G];
      return typeof r == "function" ? r : null;
    }
    var c = t.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function v(e) {
      {
        for (var r = arguments.length, n = new Array(r > 1 ? r - 1 : 0), a = 1; a < r; a++)
          n[a - 1] = arguments[a];
        k("error", e, n);
      }
    }
    function k(e, r, n) {
      {
        var a = c.ReactDebugCurrentFrame, s = a.getStackAddendum();
        s !== "" && (r += "%s", n = n.concat([s]));
        var d = n.map(function(u) {
          return String(u);
        });
        d.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, d);
      }
    }
    var N;
    N = Symbol.for("react.module.reference");
    function re(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === l || e === g || x || e === w || e === M || e === U || R || e === O || f || i || h || typeof e == "object" && e !== null && (e.$$typeof === F || e.$$typeof === I || e.$$typeof === _ || e.$$typeof === S || e.$$typeof === T || e.$$typeof === N || e.getModuleId !== void 0));
    }
    function ve(e, r, n) {
      var a = e.displayName;
      if (a)
        return a;
      var s = r.displayName || r.name || "";
      return s !== "" ? n + "(" + s + ")" : n;
    }
    function J(e) {
      return e.displayName || "Context";
    }
    function A(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && v("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case l:
          return "Fragment";
        case b:
          return "Portal";
        case g:
          return "Profiler";
        case w:
          return "StrictMode";
        case M:
          return "Suspense";
        case U:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case S:
            var r = e;
            return J(r) + ".Consumer";
          case _:
            var n = e;
            return J(n._context) + ".Provider";
          case T:
            return ve(e, e.render, "ForwardRef");
          case I:
            var a = e.displayName || null;
            return a !== null ? a : A(e.type) || "Memo";
          case F: {
            var s = e, d = s._payload, u = s._init;
            try {
              return A(u(d));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var $ = Object.assign, V = 0, K, X, m, D, Ce, xe, Se;
    function Oe() {
    }
    Oe.__reactDisabledLog = !0;
    function tt() {
      {
        if (V === 0) {
          K = console.log, X = console.info, m = console.warn, D = console.error, Ce = console.group, xe = console.groupCollapsed, Se = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: Oe,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        V++;
      }
    }
    function rt() {
      {
        if (V--, V === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: $({}, e, {
              value: K
            }),
            info: $({}, e, {
              value: X
            }),
            warn: $({}, e, {
              value: m
            }),
            error: $({}, e, {
              value: D
            }),
            group: $({}, e, {
              value: Ce
            }),
            groupCollapsed: $({}, e, {
              value: xe
            }),
            groupEnd: $({}, e, {
              value: Se
            })
          });
        }
        V < 0 && v("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var ge = c.ReactCurrentDispatcher, me;
    function ne(e, r, n) {
      {
        if (me === void 0)
          try {
            throw Error();
          } catch (s) {
            var a = s.stack.trim().match(/\n( *(at )?)/);
            me = a && a[1] || "";
          }
        return `
` + me + e;
      }
    }
    var he = !1, ae;
    {
      var nt = typeof WeakMap == "function" ? WeakMap : Map;
      ae = new nt();
    }
    function je(e, r) {
      if (!e || he)
        return "";
      {
        var n = ae.get(e);
        if (n !== void 0)
          return n;
      }
      var a;
      he = !0;
      var s = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var d;
      d = ge.current, ge.current = null, tt();
      try {
        if (r) {
          var u = function() {
            throw Error();
          };
          if (Object.defineProperty(u.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(u, []);
            } catch (W) {
              a = W;
            }
            Reflect.construct(e, [], u);
          } else {
            try {
              u.call();
            } catch (W) {
              a = W;
            }
            e.call(u.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (W) {
            a = W;
          }
          e();
        }
      } catch (W) {
        if (W && a && typeof W.stack == "string") {
          for (var o = W.stack.split(`
`), C = a.stack.split(`
`), y = o.length - 1, p = C.length - 1; y >= 1 && p >= 0 && o[y] !== C[p]; )
            p--;
          for (; y >= 1 && p >= 0; y--, p--)
            if (o[y] !== C[p]) {
              if (y !== 1 || p !== 1)
                do
                  if (y--, p--, p < 0 || o[y] !== C[p]) {
                    var j = `
` + o[y].replace(" at new ", " at ");
                    return e.displayName && j.includes("<anonymous>") && (j = j.replace("<anonymous>", e.displayName)), typeof e == "function" && ae.set(e, j), j;
                  }
                while (y >= 1 && p >= 0);
              break;
            }
        }
      } finally {
        he = !1, ge.current = d, rt(), Error.prepareStackTrace = s;
      }
      var H = e ? e.displayName || e.name : "", Ye = H ? ne(H) : "";
      return typeof e == "function" && ae.set(e, Ye), Ye;
    }
    function at(e, r, n) {
      return je(e, !1);
    }
    function it(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function ie(e, r, n) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return je(e, it(e));
      if (typeof e == "string")
        return ne(e);
      switch (e) {
        case M:
          return ne("Suspense");
        case U:
          return ne("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case T:
            return at(e.render);
          case I:
            return ie(e.type, r, n);
          case F: {
            var a = e, s = a._payload, d = a._init;
            try {
              return ie(d(s), r, n);
            } catch {
            }
          }
        }
      return "";
    }
    var oe = Object.prototype.hasOwnProperty, Pe = {}, Fe = c.ReactDebugCurrentFrame;
    function fe(e) {
      if (e) {
        var r = e._owner, n = ie(e.type, e._source, r ? r.type : null);
        Fe.setExtraStackFrame(n);
      } else
        Fe.setExtraStackFrame(null);
    }
    function ot(e, r, n, a, s) {
      {
        var d = Function.call.bind(oe);
        for (var u in e)
          if (d(e, u)) {
            var o = void 0;
            try {
              if (typeof e[u] != "function") {
                var C = Error((a || "React class") + ": " + n + " type `" + u + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[u] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw C.name = "Invariant Violation", C;
              }
              o = e[u](r, u, a, n, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (y) {
              o = y;
            }
            o && !(o instanceof Error) && (fe(s), v("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", a || "React class", n, u, typeof o), fe(null)), o instanceof Error && !(o.message in Pe) && (Pe[o.message] = !0, fe(s), v("Failed %s type: %s", n, o.message), fe(null));
          }
      }
    }
    var ft = Array.isArray;
    function be(e) {
      return ft(e);
    }
    function ut(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, n = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return n;
      }
    }
    function st(e) {
      try {
        return ke(e), !1;
      } catch {
        return !0;
      }
    }
    function ke(e) {
      return "" + e;
    }
    function Ae(e) {
      if (st(e))
        return v("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ut(e)), ke(e);
    }
    var Z = c.ReactCurrentOwner, lt = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Ie, $e, ye;
    ye = {};
    function ct(e) {
      if (oe.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function dt(e) {
      if (oe.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function vt(e, r) {
      if (typeof e.ref == "string" && Z.current && r && Z.current.stateNode !== r) {
        var n = A(Z.current.type);
        ye[n] || (v('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', A(Z.current.type), e.ref), ye[n] = !0);
      }
    }
    function gt(e, r) {
      {
        var n = function() {
          Ie || (Ie = !0, v("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: n,
          configurable: !0
        });
      }
    }
    function mt(e, r) {
      {
        var n = function() {
          $e || ($e = !0, v("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: n,
          configurable: !0
        });
      }
    }
    var ht = function(e, r, n, a, s, d, u) {
      var o = {
        $$typeof: E,
        type: e,
        key: r,
        ref: n,
        props: u,
        _owner: d
      };
      return o._store = {}, Object.defineProperty(o._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(o, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: a
      }), Object.defineProperty(o, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: s
      }), Object.freeze && (Object.freeze(o.props), Object.freeze(o)), o;
    };
    function bt(e, r, n, a, s) {
      {
        var d, u = {}, o = null, C = null;
        n !== void 0 && (Ae(n), o = "" + n), dt(r) && (Ae(r.key), o = "" + r.key), ct(r) && (C = r.ref, vt(r, s));
        for (d in r)
          oe.call(r, d) && !lt.hasOwnProperty(d) && (u[d] = r[d]);
        if (e && e.defaultProps) {
          var y = e.defaultProps;
          for (d in y)
            u[d] === void 0 && (u[d] = y[d]);
        }
        if (o || C) {
          var p = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          o && gt(u, p), C && mt(u, p);
        }
        return ht(e, o, C, s, a, Z.current, u);
      }
    }
    var pe = c.ReactCurrentOwner, De = c.ReactDebugCurrentFrame;
    function B(e) {
      if (e) {
        var r = e._owner, n = ie(e.type, e._source, r ? r.type : null);
        De.setExtraStackFrame(n);
      } else
        De.setExtraStackFrame(null);
    }
    var Ee;
    Ee = !1;
    function Re(e) {
      return typeof e == "object" && e !== null && e.$$typeof === E;
    }
    function We() {
      {
        if (pe.current) {
          var e = A(pe.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function yt(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), n = e.lineNumber;
          return `

Check your code at ` + r + ":" + n + ".";
        }
        return "";
      }
    }
    var Me = {};
    function pt(e) {
      {
        var r = We();
        if (!r) {
          var n = typeof e == "string" ? e : e.displayName || e.name;
          n && (r = `

Check the top-level render call using <` + n + ">.");
        }
        return r;
      }
    }
    function Ne(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var n = pt(r);
        if (Me[n])
          return;
        Me[n] = !0;
        var a = "";
        e && e._owner && e._owner !== pe.current && (a = " It was passed a child from " + A(e._owner.type) + "."), B(e), v('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', n, a), B(null);
      }
    }
    function Ue(e, r) {
      {
        if (typeof e != "object")
          return;
        if (be(e))
          for (var n = 0; n < e.length; n++) {
            var a = e[n];
            Re(a) && Ne(a, r);
          }
        else if (Re(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var s = Y(e);
          if (typeof s == "function" && s !== e.entries)
            for (var d = s.call(e), u; !(u = d.next()).done; )
              Re(u.value) && Ne(u.value, r);
        }
      }
    }
    function Et(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var n;
        if (typeof r == "function")
          n = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === T || r.$$typeof === I))
          n = r.propTypes;
        else
          return;
        if (n) {
          var a = A(r);
          ot(n, e.props, "prop", a, e);
        } else if (r.PropTypes !== void 0 && !Ee) {
          Ee = !0;
          var s = A(r);
          v("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", s || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && v("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Rt(e) {
      {
        for (var r = Object.keys(e.props), n = 0; n < r.length; n++) {
          var a = r[n];
          if (a !== "children" && a !== "key") {
            B(e), v("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", a), B(null);
            break;
          }
        }
        e.ref !== null && (B(e), v("Invalid attribute `ref` supplied to `React.Fragment`."), B(null));
      }
    }
    function Le(e, r, n, a, s, d) {
      {
        var u = re(e);
        if (!u) {
          var o = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (o += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var C = yt(s);
          C ? o += C : o += We();
          var y;
          e === null ? y = "null" : be(e) ? y = "array" : e !== void 0 && e.$$typeof === E ? (y = "<" + (A(e.type) || "Unknown") + " />", o = " Did you accidentally export a JSX literal instead of a component?") : y = typeof e, v("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", y, o);
        }
        var p = bt(e, r, n, s, d);
        if (p == null)
          return p;
        if (u) {
          var j = r.children;
          if (j !== void 0)
            if (a)
              if (be(j)) {
                for (var H = 0; H < j.length; H++)
                  Ue(j[H], e);
                Object.freeze && Object.freeze(j);
              } else
                v("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ue(j, e);
        }
        return e === l ? Rt(p) : Et(p), p;
      }
    }
    function wt(e, r, n) {
      return Le(e, r, n, !0);
    }
    function _t(e, r, n) {
      return Le(e, r, n, !1);
    }
    var Tt = _t, Ct = wt;
    ee.Fragment = l, ee.jsx = Tt, ee.jsxs = Ct;
  }()), ee;
}
(function(t) {
  process.env.NODE_ENV === "production" ? t.exports = cr() : t.exports = dr();
})(Ze);
const P = Ze.exports.jsx, z = /* @__PURE__ */ new Map(), se = /* @__PURE__ */ new Map(), Te = St(null);
Te.displayName = "ArrowContext";
function Qe(t) {
  const f = kt(Te);
  if (f === null) {
    const i = new Error(`<${t} /> is missing a parent <Float /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(i, Qe), i;
  }
  return f;
}
const et = Ot((t, f) => {
  var K, X;
  const i = sr(), [h, R] = le(!1), [x, E] = le(t.show !== void 0 ? t.show : !1), [b, l] = le(), w = jt(null), g = {
    show: t.onShow || (() => {
    }),
    hide: t.onHide || (() => {
    }),
    update: t.onUpdate || (() => {
    })
  }, {
    x: _,
    y: S,
    placement: T,
    strategy: M,
    reference: U,
    floating: I,
    update: F,
    refs: O,
    middlewareData: L
  } = $t({
    placement: t.placement || "bottom-start",
    strategy: t.strategy,
    middleware: b
  }), G = Pt(() => {
    F(), g.update();
  }, [F]);
  te(() => {
    const m = [];
    (typeof t.offset == "number" || typeof t.offset == "object" || typeof t.offset == "function") && m.push(Dt(t.offset)), (t.flip === !0 || typeof t.flip == "number" || typeof t.flip == "object") && m.push(Wt({
      padding: typeof t.flip == "number" ? t.flip : void 0,
      ...typeof t.flip == "object" ? t.flip : {}
    })), (t.shift === !0 || typeof t.shift == "number" || typeof t.shift == "object") && m.push(Mt({
      padding: typeof t.shift == "number" ? t.shift : void 0,
      ...typeof t.shift == "object" ? t.shift : {}
    })), (t.autoPlacement === !0 || typeof t.autoPlacement == "object") && m.push(Nt(typeof t.autoPlacement == "object" ? t.autoPlacement : void 0)), (t.arrow === !0 || typeof t.arrow == "number") && m.push(Ut({
      element: w,
      padding: t.arrow === !0 ? 0 : t.arrow
    })), m.push(...typeof t.middleware == "function" ? t.middleware({
      referenceEl: O.reference,
      floatingEl: O.floating
    }) : t.middleware || []), (t.hide === !0 || typeof t.hide == "object") && m.push(Lt(typeof t.hide == "object" ? t.hide : void 0)), l(m);
  }, [t.offset, t.shift, t.flip, t.arrow, t.autoPlacement, t.hide, t.middleware]);
  const Y = () => {
    O.reference.current && O.floating.current && t.autoUpdate !== !1 && !z.get(i) && z.set(i, Yt(O.reference.current, O.floating.current, or(G, 16), typeof t.autoUpdate == "object" ? t.autoUpdate : void 0));
  }, c = () => {
    z.get(i)(), z.delete(i);
  };
  Xe(() => (O.reference.current && O.floating.current && x === !0 && !se.get(i) ? (se.set(i, !0), g.show(), Y()) : x === !1 && se.get(i) && z.get(i) && (se.delete(i), c(), g.hide()), z.get(i)), [x]), te(() => {
    R(!0);
  }, []);
  const v = {
    arrowRef: w,
    placement: T,
    x: (K = L.arrow) == null ? void 0 : K.x,
    y: (X = L.arrow) == null ? void 0 : X.y
  }, [k, N] = t.children;
  if (!Je(k))
    return /* @__PURE__ */ P(q, {});
  const re = Ft(() => typeof t.originClass == "function" ? t.originClass(T) : typeof t.originClass == "string" ? t.originClass : t.tailwindcssOriginClass ? lr(T) : "", [t.originClass, t.tailwindcssOriginClass]), ve = {
    show: h ? t.show : !1,
    enter: `${t.enter || ""} ${re}`,
    enterFrom: `${t.enterFrom || ""}`,
    enterTo: `${t.enterTo || ""}`,
    leave: `${t.leave || ""} ${re}`,
    leaveFrom: `${t.leaveFrom || ""}`,
    leaveTo: `${t.leaveTo || ""}`,
    beforeEnter: () => {
      E(!0);
    },
    afterLeave: () => {
      E(!1);
    }
  }, J = {
    ref: I,
    style: t.transform || t.transform === void 0 ? {
      position: M,
      zIndex: t.zIndex || 9999,
      top: 0,
      left: 0,
      right: "auto",
      bottom: "auto",
      transform: `translate(${Math.round(_ || 0)}px,${Math.round(S || 0)}px)`
    } : {
      position: M,
      zIndex: t.zIndex || 9999,
      top: `${S || 0}px`,
      left: `${_ || 0}px`
    }
  }, A = (m) => {
    if (t.as === q)
      return /* @__PURE__ */ P(q, {
        children: m
      });
    const D = t.as || "div";
    return /* @__PURE__ */ P(D, {
      ref: f,
      className: t.className,
      children: m
    });
  }, $ = (m) => {
    if (h && t.portal) {
      const D = document == null ? void 0 : document.querySelector(t.portal === !0 ? "body" : t.portal);
      if (D)
        return At(m, D);
    }
    return m;
  }, V = (m) => {
    if (t.floatingAs === q)
      return /* @__PURE__ */ P(m.type, {
        ...m.props,
        ...J
      });
    const D = t.floatingAs || "div";
    return /* @__PURE__ */ P(D, {
      ...J,
      children: /* @__PURE__ */ P(m.type, {
        ...m.props
      })
    });
  };
  return A([/* @__PURE__ */ P(k.type, {
    ...k.props,
    ref: U
  }, "ReferenceNode"), /* @__PURE__ */ P(Te.Provider, {
    value: v,
    children: $(V(/* @__PURE__ */ P(It, {
      as: q,
      ...ve,
      children: /* @__PURE__ */ P(N.type, {
        ...N.props
      })
    })))
  }, "FloatingNode")]);
});
et.displayName = "Float";
function vr(t) {
  var l;
  const {
    arrowRef: f,
    placement: i,
    x: h,
    y: R
  } = Qe("Float.Arrow"), x = {
    top: "bottom",
    right: "left",
    bottom: "top",
    left: "right"
  }[i.split("-")[0]], E = {
    left: typeof h == "number" ? `${h}px` : "",
    top: typeof R == "number" ? `${R}px` : "",
    right: "",
    bottom: "",
    [x]: `${((l = t.offset) != null ? l : 4) * -1}px`
  };
  if (t.as === q) {
    const w = {
      placement: i
    }, g = typeof t.children == "function" ? t.children(w) : t.children;
    if (!g || !Je(g))
      throw new Error("When the prop `as` of <Float.Arrow /> is <Fragment />, there must be contains 1 child element.");
    return /* @__PURE__ */ P(g.type, {
      ...g.props,
      ref: f,
      style: E
    });
  }
  const b = t.as || "div";
  return /* @__PURE__ */ P(b, {
    ...t,
    ref: f,
    style: E
  });
}
const pr = Object.assign(et, {
  Arrow: vr
});
export {
  pr as Float,
  lr as tailwindcssOriginClassResolver,
  yr as tailwindcssOriginSafelist
};
